go-controller/pkg/ovn/obj_retry.go

```diff
@@ -126,18 +189,1067 @@ func (r *retryObjs) getObjRetryEntry(key string) *retryObjEntry {
+			if !isResourceScheduled(r.oType, entry.oldObj) {
+				klog.V(5).Infof("Retry: %s %s not scheduled", r.oType, objKey)
+				continue
+			}
+			if err := oc.deleteResource(r, entry.oldObj, entry.config); err != nil {
+				klog.Infof("%v retry delete failed for %s, will try again later: %v", r.oType, objKey, err)
+				entry.timeStamp = time.Now()
+				continue
+			}
+			// successfully cleaned up old object, remove it from the retry cache
+			entry.oldObj = nil
+		}
+
+		// create new object if needed
+		if entry.newObj != nil {
+			klog.Infof("%v retry: creating object for %s", r.oType, objKey)
+			if !isResourceScheduled(r.oType, entry.newObj) {
+				klog.V(5).Infof("Retry: %s %s not scheduled", r.oType, objKey)
+				continue
+			}
+			if err := oc.addResource(r, entry.newObj, true); err != nil {
+				klog.Infof("%v retry create failed for %s, will try again later: %v", r.oType, objKey, err)
+				entry.timeStamp = time.Now()
+				continue
+			}
+			// successfully cleaned up old object, remove it from the retry cache
+			entry.newObj = nil
+		}
+
+		klog.Infof("%v retry successful for %s", r.oType, objKey)
+		r.deleteRetryObj(objKey, false)
+
+	}
+}
+
+// periodicallyRetryResources tracks retryObjs and checks if any object needs to be retried for add or delete every
+// retryObjInterval seconds or when requested through retryChan.
+func (oc *Controller) periodicallyRetryResources(r *retryObjs) {
+	for {
+		select {
+		case <-time.After(retryObjInterval):
+			klog.V(5).Infof("%s s have elapsed, retrying failed objects of type %v", retryObjInterval, r.oType)
+			oc.iterateRetryResources(r, false)
+
+		case <-r.retryChan:
+			klog.V(5).Infof("Retry channel got triggered: retrying failed objects of type %v", r.oType)
+			oc.iterateRetryResources(r, true)
+
+		case <-oc.stopChan:
+			klog.V(5).Infof("Stop channel got triggered: will stop retrying failed objects of type %v", r.oType)
+			return
+		}
+	}
+}
+
+// Given a *retryObjs instance, getSyncResourcesFunc retuns the sync function for a given resource type.
+// This will be then called on all existing objects when a watcher is started.
+func (oc *Controller) getSyncResourcesFunc(r *retryObjs) (func([]interface{}), error) {
+
+	var syncRetriableFunc func([]interface{}) error
+	var syncFunc func([]interface{})
+	var name string
+
+	// If a type needs a retriable sync funcion, it will set syncRetriableFunc
+	// and will keep syncFunc=nil. For a non-retriable sync func,
+	// it will directly set syncFunc.
+	switch r.oType {
+	case factory.PodType:
+		name = "SyncPods"
+		syncRetriableFunc = oc.syncPodsRetriable
+
+	case factory.PolicyType:
+		name = "syncNetworkPolicies"
+		syncRetriableFunc = oc.syncNetworkPolicies
+
+	case factory.NodeType:
+		name = "syncNodes"
+		syncRetriableFunc = oc.syncNodesRetriable
+
+	case factory.PeerServiceTypeVar,
+		factory.PeerNamespaceAndPodSelectorTypeVar:
+		name = ""
+		syncRetriableFunc = nil
```

Comment:
this might be a bug in the old code @astoycos was this intentional? I guess we should search the gp.addressSet and remove any vips in the svc subnet that dont belong to current services in the namespace?